src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -298,17 +299,22 @@ private void AnalyzeInvocationOperation(OperationAnalysisContext operationContex
 
             private void AnalyzeNameOfOperation(OperationAnalysisContext operationContext)
             {
-                // Workaround for https://github.com/dotnet/roslyn/issues/19965
-                // IOperation API does not expose potential references to methods/properties within
-                // a bound method group/property group.
+                // 'nameof(argument)' is very commonly used for reading/writing to 'argument' in following ways:
+                //   1. Reflection based usage: See https://github.com/dotnet/roslyn/issues/32488
+                //   2. Custom/Test frameworks: See https://github.com/dotnet/roslyn/issues/32008 and https://github.com/dotnet/roslyn/issues/31581
+                // We treat 'nameof(argument)' as ValueUsageInfo.ReadWrite instead of ValueUsageInfo.NameOnly to avoid such false positives.
```

Comment:
Will "IDE0052" still be reported in any case after this change?

Reply:
Yes, it is primarily used to flags members which have write(s), but no reads. So it will still be flagged for say fields or properties that have only initializers and/or assignments.

Reply:
What I meant is this message specifically `FeaturesResources.Private_method_0_can_be_removed_as_it_is_never_invoked`

---
In reply to: [249958101](https://github.com/dotnet/roslyn/pull/32675#discussion_r249958101) [](ancestors = 249958101)

Reply:
Seems to fire for `M` below:

```csharp
class C
{
    void M() { }

    /// <summary>
    /// Another implemenentation of <see cref="M"/>
    /// </summary>
    void M2() { }
}
```

```
Severity	Code	Description
Message	IDE0052	Private method 'C.M' can be removed as it is never invoked.
Message	IDE0051	Private member 'C.M2' is unused.
```